lms/static/scripts/frontend_apps/services/index.js

```diff
@@ -0,0 +1,50 @@
+import { createContext } from 'preact';
+import { useContext } from 'preact/hooks';
+
+export { APIService } from './api';
+export { GradingService } from './grading';
+export { VitalSourceService } from './vitalsource';
+
+/**
+ * Directory of available services.
+ *
+ * @typedef ServiceMap
+ * @prop {import('./api').APIService} [api]
+ * @prop {import('./grading').GradingService} [grading]
+ * @prop {import('./vitalsource').VitalSourceService} [vitalSource]
```

Comment:
I wonder if there's a way to type this in a more abstract manner so it's less manual? Would an abstract "service" type be useful? Is it necessary to define/restrict the keys? Just thinking...

Reply:
As long as services are looked up by a string I think you'll need a static mapping of names to types like this somewhere. A more open-ended approach that occurred to me might involve looking up the service by constructor:

```js
const gradingService = useService(GradingService)
```

In this case the return type can be assumed to be an instance of `GradingService` (or a subclass) without having to look it up in a map somewhere.

Reply:
I have implemented the above change. Looking up a service now looks as above.